project/src/UI/Upgrades/UpgradeMenu.gd

```diff
@@ -17,16 +17,27 @@ func _ready() -> void:
 	cargo_button.connect("button_down", self, "select_upgrade", [Events.UpgradeChoices.CARGO])
 	mine_button.connect("button_down", self, "select_upgrade", [Events.UpgradeChoices.MINING])
 	weapon_button.connect("button_down", self, "select_upgrade", [Events.UpgradeChoices.WEAPON])
+	
+	# Connect the completion signal to the last animated child
+	$HBoxContainer.get_child($HBoxContainer.get_child_count() - 1).connect("on_ani_complete", self, "on_ani_complete")
```

Comment:
You shouldn't access and connect to nodes like that. Your code only makes sense with the comment, which is a sign you should change it. Instead of connecting like that, I would recommend turning the open() function into a coroutine with yield: you just want to wait for the last button to finish animating.

Another option is, upon touching one button in the menu while it's opening, to seek all buttons to the end of their open animation.